mesonbuild/coredata.py

```diff
@@ -946,7 +946,7 @@ def write_cmd_line_file(build_dir, options):
 
     properties = OrderedDict()
     if options.cross_file:
-        properties['cross_file'] = options.cross_file
+        properties['cross_file'] = list(map(os.path.abspath, options.cross_file))
```

Comment:
I have two concerns, first is why this isn't being done for native files as well? the second is what happens when you pass a file that isn't an absolute path, like those that are in ~/.local/share/meson/cross.

I think what you really need to do is reuse the code in mesonbuild.coredata.Coredata.__load_config_files

Reply:
Actually, the more I think abut it, all this needs to be is
```python
if options.cross_file:
    properties['cross_files'] = options.cross_file
if options.native_file:
    properties['native_files'] = options.config_file
```